rllib/evaluation/collectors/agent_collector.py

```diff
@@ -120,6 +125,36 @@ def is_empty(self) -> bool:
         """Returns True if this collector has no data."""
         return not self.buffers or all(len(item) == 0 for item in self.buffers.values())
 
+    def _check_view_requirement(self, view_requirement_name: str, data: TensorType):
+        """Warns if data does not fit the view requirement.
+
+        Should raise an AssertionError if data does not fit the view requirement in the
+        future.
+        """
+
+        if (
+            log_once(
```

Comment:
so log_once shouldn't be in a condition that we run every single time.
if there is error, we check log_once(), then logger.warning().

Reply:
Got it, changed.